.github/workflows/test.yml

```diff
@@ -33,50 +33,6 @@ jobs:
       base_ref: ${{ github.event.merge_group.base_ref || github.event.pull_request.base.ref }}
       head_ref: ${{ github.event.merge_group.head_ref || github.event.pull_request.head.ref }}
 
-  # Remove this once https://github.com/vectordotdev/vector/issues/3771 is closed.
-  # Then, modify the `cross-linux` job to run `test` instead of `build`.
-  test-linux:
-    name: Unit - x86_64-unknown-linux-gnu
-    runs-on: [linux, ubuntu-20.04-8core]
-    needs: changes
-    env:
-      CARGO_INCREMENTAL: 0
-    if: ${{ needs.changes.outputs.source == 'true' }}
-    steps:
-      - uses: actions/checkout@v3
-      - uses: actions/cache@v3
-        name: Cache Cargo registry + index
-        with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-cargo-
-      - run: sudo -E bash scripts/environment/bootstrap-ubuntu-20.04.sh
-      - run: bash scripts/environment/prepare.sh
-      - run: echo "::add-matcher::.github/matchers/rust.json"
-      - run: make test
-        env:
-          CARGO_BUILD_JOBS: 5
-      - name: Upload test results
-        run: scripts/upload-test-results.sh
-        if: always()
-
-  test-vrl:
-    name: VRL - Linux
-    continue-on-error: true
-    runs-on: [linux, ubuntu-20.04-8core]
-    needs: changes
-    if: ${{ needs.changes.outputs.source == 'true' || needs.changes.outputs.cue == 'true' }}
-    steps:
-      - uses: actions/checkout@v3
-      - run: sudo -E bash scripts/environment/bootstrap-ubuntu-20.04.sh
-      - run: bash scripts/environment/prepare.sh
-      - run: cargo vdev test-vrl
-
   checks:
     name: Checks
     runs-on: [linux, ubuntu-20.04-8core]
```

Comment:
Not specifically related to the changes here, but we also want to move up the cache step in this workflow to run before we run `prepare.sh`. As it is currently set up, the cache is being populated after we download and install the cached dependencies.
```
      - uses: actions/cache@v3
        name: Cache Cargo registry + index
        with:
          path: |
            ~/.cargo/bin/
            ~/.cargo/registry/index/
            ~/.cargo/registry/cache/
            ~/.cargo/git/db/
          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
          restore-keys: |
            ${{ runner.os }}-cargo-
      - run: sudo -E bash scripts/environment/bootstrap-ubuntu-20.04.sh
      - run: bash scripts/environment/prepare.sh
```

Reply:
good catch 👍